### PR TITLE
[Dupe and Comment] Fix selected / active line and one more bug

### PR DESCRIPTION
**Pull request by Akatmks**
Changes:  
* *Selected lines and active line will now be properly set after do and undo.*  
* *Commented line will now be properly uncommented after do.*  
    There is a bug in the previous implementation. According to [Aegisub documentation](https://web.archive.org/web/20200217233603/http://docs.aegisub.org/3.2/Automation/Lua/Subtitle_file_interface/), line is a copy of the original line in the subtitle object and not a reference. The code for it can be found [here](https://github.com/arch1t3cht/Aegisub/blob/feature/src/auto4_lua_assfile.cpp#L148-L239).  
* *aegisub.util is no longer required by do and is removed from DepCtrl's requiredModules.*  

I've done some basic tests and I think it is working as expected. Version number is left unchanged.  
Sorry for being stubborn and arguing with everyone yesterday.  
Thank you :D

### CODE REVIEW

**Comment by garret1317** [comment]
tl;dr: bug became a feature, everything else is good, thank you

> * *Commented line will now be properly uncommented after do.*  
    There is a bug in the previous implementation. According to [Aegisub documentation](https://web.archive.org/web/20200217233603/http://docs.aegisub.org/3.2/Automation/Lua/Subtitle_file_interface/), line is a copy of the original line in the subtitle object and not a reference. The code for it can be found [here](https://github.com/arch1t3cht/Aegisub/blob/feature/src/auto4_lua_assfile.cpp#L148-L239).

I'm not entirely sure that's a bug (any more).
It's definitely not what I intended when I was writing the code, but I think now I actually prefer the "broken" behaviour, so I'm going to remove the fix. Probably should've got rid of the inaccurate comments ages ago now, but oh well. Sorry for wasting your time.

<details><summary>example of the bug becoming a feature</summary>
Say if I want to add a revision inbetween the current (uncommented) one and an old (commented) one, I can just run the script without having to comment out the new one. If I'm just adding a new revision, the source line will be uncommented anyway.

(This isn't just a contrived example so i can get rid of the fix, i actually do this sometimes.)
</details>

Not using aegisub.util is much smarter, and the sel/act fixes work great. Thanks.

**Comment by garret1317** [comment]
hm, not sure why that was marked as not accepted
changes have in fact been merged

**Comment by garret1317** [comment]
maybe i should have put "resolved" in the commit message instead
sure would be nice if this stuff was documented

**Comment by Akatmks** [comment]
Thank you!  

One small issue in a [comment](https://github.com/garret1317/aegisub-scripts/blob/016ef8486d8320818879aebc3b8a6228ca28b4c5/macros/garret.dupe-and-comment.lua#L27): We are going backwards from the last selected line to the first. If the active line is bigger than the line in process, it means it is below where the line insertion takes place, and thus needs to be moved.  

---

Sorry, when looking at the code again, I realised that I wasn't doing the additions on `sel` and `act` the optimal way.  
This new method takes O(N) time instead of O(N²). It may not actually be faster when applied on one line, but it will be marginally faster when applied to more than one lines – about 1e-5 millisecond faster. If you think it isn't that much of a hassle, you may remove the current implementation and put this onto the script instead.  
```lua
if act == sel[i] then -- Instead of iterating through every items, we only modify the current item in the selection table
    sel[i] = sel[i] + i - 1 -- i - 1 means the number of items in the selection before the current
    act = sel[i] -- the active line will be equal to one of the lines in the selection, if it does, we move it together with the selection
else
    sel[i] = sel[i] + i - 1
end
```
The undo function will be the opposite:
```
if act == sel[i] then
    sel[i] = sel[i] - i + 1
    act = sel[i]
else
    sel[i] = sel[i] - i + 1
end
```
